### ISSUE TITLE
Support multiple templates up the class hierarchy?

### ISSUE THREAD

**Issue by jwendell** [jboss-openshift/ce-arq#55]
With the multi template support, we might end up in a situation like this:

```java
@Template(url=...)
class A { ... }

@Template(url=...)
class B extends A { ... }
```

Currently we are instantiating both templates. Is this the desired behavior? Or should we support only multi templates inside a `@Templates` annotation?

@rcernich Any opinion?

**Comment by errantepiphany** [comment]
I don't mind the way it is, now that I understand what is going on. One just has to be aware and be careful of naming conflicts in templates from subclasses.

**Comment by rcernich** [comment]
I think that's probably OK, although I suspect this is not what you want to do.  As the tests are designed to test a specific template, I don't think it makes sense to have @Template in the class hierarchy, i.e. I don't see why you'd extend to add tests for a given template instantiation.

I suppose you could add some logic to flag this as an error, but I don't really see the harm.

That said, this might be useful in the future if we're ever able to separate out infrastructure details in our templates, e.g. spinning up a database.  That would make more sense in a hierarchy, since multiple different products could extend from the same base DB template.

To summarize, I think we can leave the @Template logic alone for now.

**Comment by jwendell** [comment]
Thanks, Rob. That's what I think too.

We should just to make sure to get this well documented.
